### PR TITLE
Pyupgrade all the things

### PR DESCRIPTION
**Pull request by eli-schwartz**
Fixes:
- some invalid escape sequences that will become errors in future versions of python
- f-strings everywhere!

Closes #8474

### CODE REVIEW

**Review by dcbaker — APPROVED** [review]
I don't know if you want to land https://github.com/mesonbuild/meson/pull/8476 or this first, but either way this looks good.

File: test cases/unit/39 python extmodule/blaster.py [hunk 1]
```diff
@@ -10,5 +10,5 @@
     sys.exit(1)
 
 if result != 1:
-    print('Returned result {} is not 1.'.format(result))
+    print(f'Returned result {result} is not 1.')
```
Comment (eli-schwartz) [inline test cases/unit/39 python extmodule/blaster.py:13]: The shebang LIES, this is running on python2 on macOS and failing unittests.

**Comment by eli-schwartz** [comment]
Yay, the naive mostly-automated conversion flagged something that was incredibly wrong anyway. Let's merge this one first, then that one. Should be an easy(ier) rebase.

**Comment by xclaesse** [comment]
YAY! totally +1 for using fstring everywhere! 

Is it possible to have a commit that only change there string format? I'm a bit scared that in the middle of that diff there could be changes that we want to review.

**Comment by xclaesse** [comment]
Oh, it's already like that because you do a commit with --keep-percent-format before. Neat.

**Comment by eli-schwartz** [comment]
Indeed, I had the same concern. The f-strings were the bulkiest change.

**Comment by xclaesse** [comment]
Yay, thanks!

**Pull request merged**